### PR TITLE
docker, makefile, p2p main

### PR DESCRIPTION
**Pull request by y0sher**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -45,17 +46,29 @@ endif
...
 else
 	make genproto
-	go build ${LDFLAGS} -o $(CURR_DIR)/$(BINARY)
+	go build ${LDFLAGS} -o $(BIN_DIR)/$(BINARY)
 endif
 .PHONY: build
 
 hare:
-	cd cmd/hare ; go build -o $(CURR_DIR)/go-hare; cd ..
+ifeq ($(OS),Windows_NT)
```
Comment (gavraz) [inline Makefile:57]: I think we should create a func build_path(NAME) and not repeat this if else when not necessary.

File: cmd/p2p/p2p.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+		p2pApp.Initialize(cmd)
+		p2pApp.Start(cmd, args)
+	},
+}
+
+func init() {
+	cmdp.AddCommands(Cmd)
+}
+
+type P2PApp struct {
```
Comment (gavraz) [inline cmd/p2p/p2p.go:32]: Why not remove the BaseApp? Then the Hare can be composed of P2PApp. Same goes for the SpacemeshApp.

**Review by gavraz — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (antonlerner) [reply Makefile:57]: or even use the if once to assign the correct path to $(BIN_DIR)

Comment (gavraz) [reply Makefile:57]: Yeah, but you still need to push the name in the string (see the .exe for example).

File: cmd/p2p/p2p.go (hunk 2 above)
Comment (y0sher) [reply cmd/p2p/p2p.go:32]: If we'll do that it will be a problem compiling the p2p app as a standalone

Comment (y0sher) [reply cmd/p2p/p2p.go:32]: Because it has to have a `main` to do so. and if we'll do it we won't be able to use it embedded in other structs.

**Review by gavraz — CHANGES_REQUESTED** [review]
1. Fix makefile (extract func)
2. Remove p2p dockerfile

**Comment by y0sher** [comment]
opened #660 about reducing code duplication in the Makefile

**Review by gavraz — APPROVED** [review]

**Pull request merged**